pkg/daemon/rpm-ostree.go

```diff
@@ -245,17 +297,117 @@ func (r *RpmOstreeClient) Rebase(imgURL, osImageContentDir string) (changed bool
 	args := []string{"rebase", "--experimental", fmt.Sprintf("%s:%s", repo, ostreeCsum),
 		"--custom-origin-url", customURL, "--custom-origin-description", "Managed by machine-config-operator"}
 
-	var out []byte
-	if out, err = runGetOut("rpm-ostree", args...); err != nil {
-		// capture stdout output as well in case of error
-		err = errors.Wrapf(err, "with stdout output: %v", string(out))
-		return
-	}
-
+	_, err = runRpmOstree(args[0], args[1:]...)
 	changed = true
 	return
 }
 
+// constant for describing kernel argument operations
+const (
+	kargRemove int = iota
+	kargAdd
+)
+
+// KernelArgument describes an operation for karg handling
+type KernelArgument struct {
+	Operation int
+	Name      string
+}
+
+// runRpmOStree wraps the rpm-ostree command. Unless mocked, this
+// is the rpmOstreeCommander that is used by RpmOstreeClient.
+func runRpmOstree(noun string, args ...string) ([]byte, error) {
+	errNotEnoughArgs := errors.New("rpm-ostree command does not have enough arguments")
+	errTooManyArgs := errors.New("rpm-ostree command called with too many arguments")
+
+	// Perform some seriously basic command validation on the rpm-ostree
+	// commands. These checks are not meant to be exhaustive.
+	switch noun {
+	// Argless commands
+	case "cancel", "rollblack":
+		if len(args) != 0 {
+			return nil, errTooManyArgs
+		}
+
+	// Check that "kargs" is either a query or a supported append/delete command.
+	case "kargs":
+		for _, v := range args {
+			// the MCD only supports append or delete, not replace
+			if strings.HasPrefix(v, "--append=") || strings.HasPrefix(v, "--delete") {
+				continue
+			}
+			return nil, fmt.Errorf("%s is an invalid rpm-ostree karg directive", v)
+		}
+
+	// Dumb checks for package operations.
+	case "override", "uninstall", "install":
+		if len(args) == 0 {
+			return nil, errNotEnoughArgs
+		}
+
+	// known error conditions
+	case "":
+		return nil, errNotEnoughArgs
+	default:
```

Comment:
"status" is still not supported in this switch case right?

Reply:
Yeah with this PR I get `Failed to initialize single run daemon: error reading osImageURL from rpm-ostree: unsupported command 'rpm-ostree status'`

Reply:
I reworked this to include full command validation, but this will need some serious testing.